qa/L0_shared_memory/shared_memory_test.py

```diff
@@ -29,124 +29,143 @@
 import sys
 
 sys.path.append("../common")
-
 import os
 import unittest
+from pathlib import Path
 
 import numpy as np
 import test_util as tu
 import tritonclient.grpc as grpcclient
 import tritonclient.http as httpclient
 import tritonclient.utils.shared_memory as shm
+import util
+from parameterized import parameterized
 from tritonclient import utils
 
 
 class SharedMemoryTest(tu.TestResultCollector):
-    def test_invalid_create_shm(self):
-        # Raises error since tried to create invalid system shared memory region
-        try:
-            shm_op0_handle = shm.create_shared_memory_region(
-                "dummy_data", "/dummy_data", -1
-            )
-            shm.destroy_shared_memory_region(shm_op0_handle)
-        except Exception as ex:
-            self.assertTrue(str(ex) == "unable to initialize the size")
+    # Constant members
+    shared_memory_test_client_log = Path(os.getcwd()) / "client.log"
+    model_dir_path = Path(os.getcwd()) / "models"
+    model_source_path = Path(os.getcwd()).parents[0] / "python_models/add_sub/model.py"
+    model_config_source_path = (
+        Path(os.getcwd()).parents[0] / "python_models/add_sub/config.pbtxt"
+    )
 
-    def test_valid_create_set_register(self):
-        # Create a valid system shared memory region, fill data in it and register
-        if _protocol == "http":
-            triton_client = httpclient.InferenceServerClient(_url, verbose=True)
+    # Custom setup method to allow passing of parameters
+    def _setUp(self, protocol, log_file_path):
+        self._tritonserver_ipaddr = os.environ.get("TRITONSERVER_IPADDR", "localhost")
```

Comment:
Does this need to be configurable in practice? Do we expect to use shared memory for anything other than a co-located server on localhost?

Reply:
TBD: Currently on the Windows testing side of things, it's passed in as a variable and is different from "localhost". Still trying to get a CI pipeline up to see the new behavior for this test in particular. Will remove if no issue.